### PR TITLE
Gmail no inbox failure

### PR DESCRIPTION
**Pull request by orhovy**
Added try and catch when trying to get mailboxes of an account that has no mailboxes
When one of the gmail users mailbox is disabled the command failed with an exception that `Mail service not enabled`.
This can be reproduced using this steps: https://support.google.com/a/answer/57919?hl=en

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34977

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 1]
```diff
@@ -1125,7 +1126,13 @@ def search(user_id, subject='', _from='', to='', before='', after='', filename='
         'v1',
         ['https://www.googleapis.com/auth/gmail.readonly'],
         command_args['userId'])
-    result = service.users().messages().list(**command_args).execute()
+    try:
+        result = service.users().messages().list(**command_args).execute()
+    except HTTPError as e:
```
Comment (bakatzir) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1131]: I think that this way we would miss all of the mesasges.

Comment (bakatzir) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:1131]: I would suggest try getting the messages per user, and skip the users without a mail service.

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 2]
```diff
@@ -1125,7 +1126,13 @@ def search(user_id, subject='', _from='', to='', before='', after='', filename='
...
+        if "Mail service not enabled" in str(e):
+            result = {}
+        else:
+            raise Exception(e)
```
Comment (yaakovi) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1135]:
```suggestion
            raise
```

**Review by yaakovi — COMMENTED** [review]
looks good to me, were you able to test it?

File: Packs/Gmail/Integrations/Gmail/Gmail.py (hunk 1 above)
Comment (yaakovi) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:1131]: this function searches in a specific mailbox, there is an iteration on all mailboxes in the calling function.

**Comment by orhovy** [comment]
Tested it and saw the fix.

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 3]
```diff
@@ -1125,7 +1125,13 @@ def search(user_id, subject='', _from='', to='', before='', after='', filename='
         'v1',
         ['https://www.googleapis.com/auth/gmail.readonly'],
         command_args['userId'])
-    result = service.users().messages().list(**command_args).execute()
+    try:
+        result = service.users().messages().list(**command_args).execute()
```
Comment (bakatzir) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1129]: let's log the disabled mailbox details(if we have them).

**Review by bakatzir — APPROVED** [review]

File: Packs/Gmail/Integrations/Gmail/Gmail.py (hunk 3 above)
Comment (orhovy) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:1129]: We dont have it ... the api do not mention that on the response:
HttpError: <HttpError 400 when requesting https://gmail.googleapis.com/gmail/v1/users/gsuiteroot%40wealthsimple.com/messages?q=from%3Aykornyk%40gmail.com++subject%3ATEST&alt=json&maxResults=100&includeSpamTrash=true returned "Mail service not enabled". Details: "Mail service not enabled">.
As the command returns all mailboxes we do not which one returns an error.

**Pull request merged**